# Barrnap

**1. Barrnap**

[Barrnap](https://github.com/tseemann/barrnap) (BAsic Rapid Ribosomal RNA Predictor) predicts the location of ribosomal RNA genes in genomes. It supports bacteria (5S,23S,16S), archaea (5S,5.8S,23S,16S), metazoan mitochondria (12S,16S) and eukaryotes (5S,5.8S,28S,18S). Barrnap takes FASTA DNA sequence as input and can be used on both the assembled contigs (community) and MAGs. 


a) Add the name of the sample at the beginning of every contig and change the extension type to `.fna` (same as .fasta, we are simply using a different extension to distinguish new file with the sample name in front of each contig from the old file). 
```{bash, eval=FALSE}
for i in *.fasta ; do  perl -lne 'if(/^>(\S+)/){ print ">$ARGV $1"} else{ print }' $i > $i.fna ; done
```

b) Change the space between the name and the scaffhold number to an underscore
```{bash, eval= FALSE}
sed -i 's/ /_/g' *.fna
```

c) Run Barrnap. Create a bash script called `run_barrnap.sh` with the following commands and execute the script using nohup.
```{bash, eval=FALSE}
#!/bin/bash 
for i in *.fna ; do barrnap $i > $i.barrnap_hits.gff --threads 20 ; done
```

- **Output** : For each given file Barrnap generates a GFF file as output which includes the coordinates of where the rRNA genes are encoded. 
- To extract the ribosomal sequences as FASTA we are using the function `getfasta` from `bedtools`. 

**2. Extract sequences**

The [bedtools](https://bedtools.readthedocs.io/en/latest/) utilies are a swiss-army knife of tools for a wide-range of genomics analysis tasks. Here we use the `getfasta` command to extract sequences from a FASTA file for each of the intervals defined in the GFF3 file generated by Barrnap. 

``````{bash, eval=FALSE}
#!/bin/bash 
for i in *.fna ; do bedtools getfasta -fi $i -bed $i.barrnap_hits.gff -fo $i.out_rRNA.fasta ; done
```

Transfer the `.out_rRNA.fasta` to your local computer and use NCBI-BLAST to classify the rRNA sequences identified by Barrnap. 

