--- 
title: "Metagenomic pipeline"
author: "Karine Villeneuve"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: 
  bookdown::gitbook:
    config:
      sharing: null
    css: toc.css
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: rstudio/bookdown-demo
---

# Introduction

This guide assumes the following : 

(1) The user has access to the server Titan from the Lazar lab (See section [Remote login] for more details).  
(2) The user has an active VPN access. 
(3) The paired-end fastq files from Illumina Miseq sequencing were transferred to the user's home directory.
(4) The user has followed the `Introduction to linux` guide and is comfortable with basic command line functions such as : 

    - listing files inside a current directory (`ls`) ;
    - moving from one directory to the other (`cd`) ; 
    - creating new directory (`mkdir`) ; and 
    - moving / copying (`mv`/`cp`) files from one directory to the other. 

## Setting up your environment

Keeping your files organized is a skill that has a high long-term payoff. As you are in the thick of an analysis, you may underestimate how many files/folders you have floating around. But a short time later, you may return to your files and realize your organization was not as clear as you hoped, which can ultimately lead to significantly slower research progress. Furthermore, one must keep in mind that someone unfamiliar with your project should be able to look at your computer files and understand in detail what you did and why. 

While there’s a lot of ways to keep your files organized, and there’s not a “one size fits all” organizational solution, below we propose a simple organizational scheme which is project-oriented, maintainable and ultimately follows consistent patterns for amplicon sequence analysis. Please note that the proposed workflow assumes such organization. 

```
┌─ ~ -------------------------------- Your home directory
│   ├── chapter1_metagenomics_aquifer ------ Project with a short but meaningfull name 
│       ├── 01_raw_data --------------- Raw files (i.e. fastq files generated by sequencing)
│       ├── 02_pre-assembly -----------
│       ├── 03_assembly ---------------
│       ├── 04_filtered_contigs -------
│       ├── 05_ -------
------------------ Project-specific final datasets for publication
│       ├── 03_assembly
│       ├── figures ----------------- Project-specific final figures for publication
│       └── scripts ----------------- Project-specific scripts
└──────────────────────────────────────────────────────────────────
``` 

Further reading about organizing files and foldes : 

- [Organizing your project](https://jhudatascience.org/Reproducibility_in_Cancer_Informatics/organizing-your-project.html) by the Johns Hopkins Data Science Lab

- [A Quick Guide to Organizing Computational Biology Projects](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1000424#s9) by William Stafford Noble, 2009

- [Reddit post](https://www.reddit.com/r/bioinformatics/comments/3rdlhf/help_how_do_you_organize_your_projectsfiles/)

- [Organizing your data](https://www.mdc-berlin.de/research-data-management/organizing-your-data) by The Max Delbrück Center 

## Download fastqs files from Illumina BaseSpace Sequence Hub 

1. Open the link found in the email sent by Geneviève Bourret. 

    If this is your first time downloading your fastqs create a new BaseSpace Sequence Hub account using the email address to         which the email from Geneviève was addressed (normally this would be your UQAM's email)

2. On the pop-up window informing you that the CERMO-FC has shared the following item with you click `ACCEPT`. 

3. Click on the `PROJECTS` tab in the upper section of the page. 

4. Select your project and then click on the second round logo from the left which looks like a blank page and in the drop-down menu select `DOWNLOAD` then `PROJECT`. 

5. If required, download the Illumina Basespace downloader by clicking `INSTALL DOWNLOAD` and follow the instructions. Otherwise simply click `DOWNLOAD` to begin downloading your fastqs. 

6. Once the download is complete you will find inside the folder a folder for each of your sample inside which the forward and reverse read are both found in another folder. Instead of going into each folder individually and copying the fastqs manually we can use the terminal to do the job for us. From a new local terminal window navigate to the folder containing all the folders and execute the following command after having modified `/path/to/directory/where/to/move/fastqs` to the actual path where you wish to move your fastqs. 

```{bash, eval=FALSE}
find ./ -name "*.gz" -exec cp -prv "{}" "/path/to/directory/where/to/move/fastqs" ";"
```

7. Finally you can transfer your fastqs to your assigned server using any File Transfer Protocol (FTP) clients (such as [FileZilla](https://filezilla-project.org/) or [Cyberduck](https://cyberduck.io/)) or using the SCP (secure copy) command-line utility. 

    For SCP you can copy an entire folder by opening a new local terminal window and navigating to the directory containing the folder with the fastqs. From that directory execute the following command. You will then be asked to enter the password for your  user on the server. 

```{bash, eval=FALSE}
scp -r name_of_foler_with_fastqs username@server.bio.uqam.ca:/path/to/copy/folder
```
