<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Metabolic pathway | Protocol for metagenomic analyses</title>
  <meta name="description" content="6 Metabolic pathway | Protocol for metagenomic analyses" />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Metabolic pathway | Protocol for metagenomic analyses" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Metabolic pathway | Protocol for metagenomic analyses" />
  
  
  

<meta name="author" content="Karine Villeneuve" />


<meta name="date" content="2024-08-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="binning.html"/>
<link rel="next" href="phylogenetic-tree.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Metagenomic sequencing data analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="pre-process.html"><a href="pre-process.html"><i class="fa fa-check"></i><b>2</b> Pre-process</a></li>
<li class="chapter" data-level="3" data-path="assembly-spades.html"><a href="assembly-spades.html"><i class="fa fa-check"></i><b>3</b> Assembly SPAdes</a></li>
<li class="chapter" data-level="4" data-path="barrnap.html"><a href="barrnap.html"><i class="fa fa-check"></i><b>4</b> Barrnap</a></li>
<li class="chapter" data-level="5" data-path="binning.html"><a href="binning.html"><i class="fa fa-check"></i><b>5</b> Binning</a>
<ul>
<li class="chapter" data-level="5.0.1" data-path="binning.html"><a href="binning.html#metabat2"><i class="fa fa-check"></i><b>5.0.1</b> MetaBAT2</a></li>
<li class="chapter" data-level="5.1" data-path="binning.html"><a href="binning.html#installing-checkm"><i class="fa fa-check"></i><b>5.1</b> Installing Checkm</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="metabolic-pathway.html"><a href="metabolic-pathway.html"><i class="fa fa-check"></i><b>6</b> Metabolic pathway</a></li>
<li class="chapter" data-level="7" data-path="phylogenetic-tree.html"><a href="phylogenetic-tree.html"><i class="fa fa-check"></i><b>7</b> Phylogenetic tree</a></li>
<li class="chapter" data-level="8" data-path="using-nohup-and-other-tips-and-tricks.html"><a href="using-nohup-and-other-tips-and-tricks.html"><i class="fa fa-check"></i><b>8</b> Using nohup and other tips and tricks</a>
<ul>
<li class="chapter" data-level="8.1" data-path="using-nohup-and-other-tips-and-tricks.html"><a href="using-nohup-and-other-tips-and-tricks.html#bash-scripting-using-nohup"><i class="fa fa-check"></i><b>8.1</b> Bash scripting using nohup</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="other-tips-and-tricks.html"><a href="other-tips-and-tricks.html"><i class="fa fa-check"></i><b>9</b> Other tips and tricks</a>
<ul>
<li class="chapter" data-level="9.1" data-path="other-tips-and-tricks.html"><a href="other-tips-and-tricks.html#moving-files-using-scp"><i class="fa fa-check"></i><b>9.1</b> Moving files using SCP</a></li>
<li class="chapter" data-level="9.2" data-path="other-tips-and-tricks.html"><a href="other-tips-and-tricks.html#renaiming-files"><i class="fa fa-check"></i><b>9.2</b> Renaiming files</a></li>
<li class="chapter" data-level="9.3" data-path="other-tips-and-tricks.html"><a href="other-tips-and-tricks.html#extract-scaffhold-based-on-nane-and-coordinates-using-samtools"><i class="fa fa-check"></i><b>9.3</b> Extract scaffhold based on nane and coordinates using samtools</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Protocol for metagenomic analyses</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="metabolic-pathway" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">6</span> Metabolic pathway<a href="metabolic-pathway.html#metabolic-pathway" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>1. BlastKOALA</strong></p>
<p><a href="https://www.kegg.jp/blastkoala/">BlastKOALA</a> is an automatic annotation servers for genome and metagenome sequences, which perform KO (KEGG Orthology) assignments to characterize individual gene functions and reconstruct KEGG pathways, BRITE hierarchies and KEGG modules to infer high-level functions of the organism or the ecosystem.</p>
<p>BlastKOALA takes as input amino acid sequences in FASTA format. We therefore use <a href="https://github.com/hyattpd/Prodigal">prodigal</a> to translate nucleic acid sequences to the corresponding peptide sequences.By default prodigal takes as input FASTA files with the <code>.fna</code> extension.</p>
<ol style="list-style-type: lower-alpha">
<li>Use perl to add the name of the sample to the beginning of every file and change the file type to <code>.fna</code>.</li>
</ol>
<div class="sourceCode" id="cb40"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb40-1"><a href="metabolic-pathway.html#cb40-1" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="pp">*</span>.fa <span class="kw">;</span> <span class="cf">do</span>  <span class="fu">perl</span> <span class="at">-lne</span> <span class="st">&#39;if(/^&gt;(\S+)/){ print &quot;&gt;$ARGV $1&quot;} else{ print }&#39;</span> <span class="va">$i</span> <span class="op">&gt;</span> <span class="va">$i</span>.fna <span class="kw">;</span> <span class="cf">done</span></span></code></pre></div>
<ol start="2" style="list-style-type: lower-alpha">
<li>Replace space between the name of you bins and the scaffhold to underscores</li>
</ol>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb41-1"><a href="metabolic-pathway.html#cb41-1" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="pp">*</span>.fna <span class="kw">;</span> <span class="cf">do</span> <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/ /_/g&#39;</span> <span class="va">$i</span> <span class="kw">;</span> <span class="cf">done</span> </span></code></pre></div>
<ol start="3" style="list-style-type: lower-alpha">
<li>Run <code>prodigal</code></li>
</ol>
<div class="sourceCode" id="cb42"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb42-1"><a href="metabolic-pathway.html#cb42-1" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="pp">*</span>.fna <span class="kw">;</span> <span class="cf">do</span> <span class="ex">prodigal</span> <span class="at">-i</span> <span class="va">$i</span> <span class="at">-o</span> output.txt <span class="at">-a</span> <span class="va">$i</span>.faa <span class="kw">;</span> <span class="cf">done</span></span></code></pre></div>
<ol start="4" style="list-style-type: lower-alpha">
<li>Remove all the characters after the first space in the header</li>
</ol>
<div class="sourceCode" id="cb43"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb43-1"><a href="metabolic-pathway.html#cb43-1" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="pp">*</span>.faa<span class="kw">;</span> <span class="cf">do</span> <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/\s.*$//&#39;</span> <span class="va">$i</span><span class="kw">;</span> <span class="cf">done</span> </span></code></pre></div>
<ol start="5" style="list-style-type: lower-alpha">
<li>Transfer the files to your local computer and follow the instructions on the BlastKOALA website for submission.</li>
</ol>
<p><strong>2. Other programs we aren’t using anymore</strong></p>
<ol style="list-style-type: decimal">
<li>DRAM</li>
</ol>
<p><a href="https://github.com/WrightonLabCSU/DRAM">DRAM</a> (Distilled and Refined Annotation of Metabolism) is a tool for annotating metagenomic assembled genomes and VirSorter identified viral contigs. DRAM annotates MAGs and viral contigs using KEGG (if provided by the user), UniRef90, PFAM, dbCAN, RefSeq viral, VOGDB and the MEROPS peptidase database as well as custom user databases.</p>
<ol style="list-style-type: lower-alpha">
<li>Activate environment</li>
</ol>
<div class="sourceCode" id="cb44"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb44-1"><a href="metabolic-pathway.html#cb44-1" tabindex="-1"></a><span class="ex">conda</span> activate DRAM</span></code></pre></div>
<ol start="2" style="list-style-type: lower-alpha">
<li>Execute DRAM. Create a bash script called <code>run_dram.sh</code> with the following commands and execute the script using nohup.Note : DRAM requires the full path (home/genomics/…) and won’t accept path starting from the home directory (~).</li>
</ol>
<div class="sourceCode" id="cb45"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb45-1"><a href="metabolic-pathway.html#cb45-1" tabindex="-1"></a><span class="ex">DRAM.py</span> annotate <span class="at">-i</span> <span class="st">&#39;/home/genomics/yourname/samples/*.fa&#39;</span> <span class="at">-o</span> /home/genomics/yourname/samples/annotation <span class="at">--threads</span> 40 </span></code></pre></div>
<ol start="3" style="list-style-type: lower-alpha">
<li>Once annotation is done, the following command will summarize all the results from the folder <code>annotate</code></li>
</ol>
<div class="sourceCode" id="cb46"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb46-1"><a href="metabolic-pathway.html#cb46-1" tabindex="-1"></a><span class="ex">DRAM.py</span> distill <span class="at">-i</span> annotation/annotations.tsv <span class="at">-o</span> summaries <span class="at">--trna_path</span> annotation/trnas.tsv <span class="at">--rrna_path</span> annotation/rrnas.tsv</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Metabolic</li>
</ol>
<p><a href="https://github.com/AnantharamanLab/METABOLIC">METABOLIC</a> (METabolic And BiogeOchemistry anaLyses In miCrobes) enables the prediction of metabolic and biogeochemical functional trait profiles to any given genome datasets.</p>
<ol style="list-style-type: lower-alpha">
<li>Activate environment</li>
</ol>
<div class="sourceCode" id="cb47"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb47-1"><a href="metabolic-pathway.html#cb47-1" tabindex="-1"></a><span class="ex">conda</span> activate metabolic</span></code></pre></div>
<ol start="2" style="list-style-type: lower-alpha">
<li>METABOLIC takes as input amino acid sequences in FASTA format. Create a bash script called <code>run_metabolic.sh</code> with the following commands and execute the script using nohup. Before running update the path to reflect the actual path where your amino acid FASTA files are located and where you want METABOLIC to store the output files.</li>
</ol>
<div class="sourceCode" id="cb48"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb48-1"><a href="metabolic-pathway.html#cb48-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb48-2"><a href="metabolic-pathway.html#cb48-2" tabindex="-1"></a><span class="fu">perl</span> /home/genomics/METABOLIC/METABOLIC-G.pl <span class="at">-in</span> /home/genomics/user/07_metabolic_pathway/ <span class="at">-o</span> /home/genomics/user/07_metabolic_pathway/metabolic_outut/</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>MEBS</li>
</ol>
<p>[MEBS] (<a href="https://github.com/valdeanda/mebs" class="uri">https://github.com/valdeanda/mebs</a>)(Multigenomic Entropy-Based Score) allows the user to synthesizes genomic information into a single informative value. This entropy score can be used to infer the likelihood that microbial taxa perform specific metabolic-biogeochemical pathways.</p>
<p>For the moment MEBS shoudl be installed on the user’s local computer. See <a href="https://valdeanda.github.io/mebs/README-1.html">manual</a> for installation and instructions.</p>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="binning.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="phylogenetic-tree.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/05_metabolic_patways.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
